Android WebViews documentation update (#904)
apache/cordova-docs: www/docs/en/dev/guide/platforms/android/webview.md

   [cordova.apache.org](http://cordova.apache.org) and unzip its
   Android package.

> 

Reviewer: Does this still apply?

Author: Config.getStartUrl() is marked as deprecated and has changed. I didn't touch the camera but assume it has changed as well.

> 

Reviewer: What about this?

Author: Again, not sure if this still applicable. I can look into this when I work on the plugin for the native-to-cordova-bridge

> 1. Follow the instructions here to build your first cordova app [CreateYourFirstApp](https://cordova.apache.org/docs/en/8.x/guide/cli/index.html)

Reviewer: ```suggestion
1. Follow [these instructions](https://cordova.apache.org/docs/en/latest/guide/cli/index.html) to build your first Cordova app
```

> 1. Follow the instructions here to build your first cordova app [CreateYourFirstApp](https://cordova.apache.org/docs/en/8.x/guide/cli/index.html)

Reviewer: Why does one do that? Just to get the files to your harddisk from where you can copy them?

Author: My interpretation (and use case) was to embed a cordova app into another Android app that was developed with native code. As this is a getting started documentation, I assumed it's a good starting point to build a first app. Everyone having a working cordova app by hand will obviously skip the step. Does this make sense?

Reviewer: Almost. What do you need the Cordova app for when actually embedding the Cordova Android webview in the native app?

Author: One of my customers has the requirement to develop two native apps within a larger project. He also needs to have geospatial capabilities (like Google maps) but without the restrictions of the popular SDK's provided by Mapbox, Google,... The organization that will be offering the apps has licenses for the "Hexagon Mapworks SDK". But this is more a ready-to-use app than a SDK so I needed to run it within the native target app.

Author: What I've seen and learned is that the cordova build applies some magic to the html sources when copying them into the platform(s). I didn't look into this further but without a build for android, no assets/www sources are available.
The same applies to all other files and folders as only the full cordova build (cordova create + cordova add platform + cordova build) creates the android project sources.

So the scenario(s) are two different ones:
1. User has a native app AND a working cordova app that needs to be integrated. (Described here)
2. User has nothing, wants to create a cordova app and then do some native work in the app

With two different starting points, it's quite difficult to described it in general :-)

Reviewer: Yep - I have never build an app that uses Cordova in that way at all, that's why I am asking such dumb questions.

1. Copy Cordova framework files

* `platforms/android/CordovaLib/src/.*` to `<AndroidAppRoot>/app/main/java/`
* `platforms/android/src/*` to `<AndroidAppRoot>/app/main/java/`

1. Modify the layout file of the activity that shall host the Cordova view e.g.
```
<LinearLayout android:layout_width="fill_parent"
              android:layout_height="fill_parent"
              android:orientation="vertical"
              xmlns:android="http://schemas.android.com/apk/res/android">
    <org.apache.cordova.engine.SystemWebView
                android:id="@+id/cordovaWebView"
                android:layout_width="match_parent"
                android:layout_height="match_parent" />
</LinearLayout
```

> 1. Modify your activity so that the class extends CordovaActivity (found at app/main/java/org/apache/cordova/CordovaActivity.java)

Reviewer: ```suggestion
1. Modify your activity so that the class extends `CordovaActivity` (found at `app/main/java/org/apache/cordova/CordovaActivity.java`)
```

```
public class TestActivity extends CordovaActivity {
    
}
```
   
1. Override `onCreate`, `makeWebView` and `createViews` to use your defined layout
```
@Override
public void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);

>      setContentView(R.layout.activity_test); //this is the layout file for your activity

Reviewer: ```suggestion
     setContentView(R.layout.activity_test); // this is the layout file for your activity
```

     super.init();
     loadUrl(launchUrl);
}

@Override
protected CordovaWebView makeWebView() {

>     SystemWebView appView = (SystemWebView) findViewById(R.id.cordovaWebView);  //this is the id for the SystemWebView in step 4

Reviewer: ```suggestion
    SystemWebView appView = (SystemWebView) findViewById(R.id.cordovaWebView);  // id for the SystemWebView in previous
```

   [cordova.apache.org](http://cordova.apache.org) and unzip its
   Android package.

1. Follow the instructions here to build your first cordova app [CreateYourFirstApp](https://cordova.apache.org/docs/en/8.x/guide/cli/index.html)

1. Copy Cordova framework files

> ```

Reviewer: this should probably be a normal list (with *) and then ` to highlight the paths, but "to" normal text

> ```

Author: ```suggestion
*
```

Author: ```suggestion
```

> platforms/android/CordovaLib/src/.* to <AndroidAppRoot>/app/main/java

Author: ```suggestion
`platforms/android/CordovaLib/src/.*` to `<AndroidAppRoot>/app/main/java/`
```

Author: ```suggestion
* `platforms/android/CordovaLib/src/.*` to `<AndroidAppRoot>/app/main/java/`
```

> platforms/android/src/** to <AndroidAppRoot>/app/main/java

Author: ```suggestion
* `platforms/android/src/.*` to `<AndroidAppRoot>/app/main/java/`
```

> ```

Author: ```suggestion
```

1. Modify the layout file of the activity that shall host the Cordova view e.g.
```
<LinearLayout android:layout_width="fill_parent"
              android:layout_height="fill_parent"
              android:orientation="vertical"
              xmlns:android="http://schemas.android.com/apk/res/android">
    <org.apache.cordova.engine.SystemWebView
                android:id="@+id/cordovaWebView"
                android:layout_width="match_parent"
                android:layout_height="match_parent" />
</LinearLayout
```

> 1. Modify your activity so that the class extends CordovaActivity (found at app/main/java/org/apache/....CordovaActivity.java)

Reviewer: ```suggestion
1. Modify your activity so that the class extends `CordovaActivity` (found at `app/main/java/org/apache/....CordovaActivity.java`)
```

What does the `....` mean here?

Author: ```suggestion
1. Modify your activity so that the class extends CordovaActivity (found at app/main/java/org/apache/cordova/CordovaActivity.java)
```
I'm using "..." as a placeholder in my docs. Now it's the full path

```
public class TestActivity extends CordovaActivity {
    
}
```
   

> 1. Override onCreate, makeWebView and createViews to use your defined layout

Reviewer: ```suggestion
1. Override `onCreate`, `makeWebView` and `createViews` to use your defined layout
```

Author: ```suggestion
1. Override `onCreate`, `makeWebView` and `createViews` to use your defined layout
```
Got it...
